src/Target.cpp

```diff
@@ -753,6 +799,88 @@ int Target::natural_vector_size(const Halide::Type &t) const {
     }
 }
 
+bool Target::get_runtime_compatible_target(const Target& other, Target &result) {
```

Comment:
Since at least one caller calls with 'this' and 'result' referring to the same instance, we should document that as supported behavior, and double-check this code to be sure there are no issues. (Might be simplest to construct the result in a local tmp and assign to 'result' at the end, since this object is small and cheap.)

Reply:
@steven-johnson - done! just pushed :)

Reply:
It's too bad that `std::optional` isn't available in C++11, though! That would certainly be a much cleaner interface to this function.

Reply:
I suspect we'll be able to move to C++17 within a year or so :-)